lib/services/restaurant_search_api.dart

```diff
@@ -1,27 +1,145 @@
+  var address = Address.fromJson(userAddress);
+  String userLocation = address.toString();
+  return userLocation;
+}
+
+class Address {
+  final String si;
+  final String gu;
+  final String dong;
+  final String detail1;
+  final String detail2;
+
+  Address(this.si, this.gu, this.dong, this.detail1, this.detail2);
+
+  Address.fromJson(Map<String, dynamic> json)
+      : si = json["results"][0]['region']['area1']['name'],
+        gu = json["results"][0]['region']['area2']['name'],
+        dong = json["results"][0]['region']['area3']['name'],
+        detail1 = json["results"][0]['land']['number1'],
+        detail2 = json["results"][0]['land']['number2'];
+
+  Map<String, dynamic> toJson() => {
+        'si': si,
+        'gu': gu,
+        'dong': dong,
+        'detail1': detail1,
+        'detail2': detail2,
+      };
+
+  @override
+  String toString() {
+    return si + " " + gu + " " + dong + " " + detail1 + "-" + detail2;
+  }
+}
+
+Future<dynamic> getRestaurantList(String gusi) async {
+  debugPrint(gusi);
+  List<RestaurantModel> restaurantList = [];
+  List<String> foodList = [
+    "맛집",
+    "고깃",
+    "햄버거",
+    "해산물",
+    "족발",
+    "보쌈",
+    "피자",
+    "치킨",
+    "국밥",
+    "분식"
+  ];
+
+  for (int i = 0; i < foodList.length; i++) {
+    final uri = Uri.parse(
+        "https://openapi.naver.com/v1/search/local.json?query=$gusi ${foodList[i]}&display=20&start=1&sort=random");
+    final Map<String, String> header = {
+      "X-Naver-Client-Id": "uWDv77C8E2CmY5bUBJ92",
+      "X-Naver-Client-Secret": "aSgxIylTMz",
+      "display": "1"
+    };
+
+    final response = await http.get(uri, headers: header);
+    //debugPrint(response.body);
+    String jsonRestaurantData = response.body;
+    final restaurantJsonList =
+        (jsonDecode(jsonRestaurantData)["items"] as List<dynamic>);
 
-const String _kNativeAppKey = 'db88ffc1aed093e91db845b457d49495';
-const String _kJavaScriptKey = '7353032fb3c902fbc319097fbc35a44a';
-const String _kRestApiKey = '665888240dc5aa6f10d609f5be6c20d1';
-
-Future<List> searchRestaurant() async {
-  var url = Uri.parse(
-      'https://dapi.kakao.com/v2/maps/sdk.js?appkey=$_kJavaScriptKey&libraries=services');
-  var response = await http.get(url);
-
-  final flutterJs = getJavascriptRuntime();
-  final result = flutterJs.evaluate('''
-script.src = '//dapi.kakao.com/v2/maps/sdk.js?appkey=$_kNativeAppKey&libraries=services&autoload=false';
-script.onload = () => {
-  kakao.maps.load(() => {
-    console.assert(kakao.maps.Map);
-  });
-};
-''');
-  print(result.stringResult);
-  return [];
+    for (final model in restaurantJsonList) {
+      restaurantList
+          .add(RestaurantModel.fromNaverApiJson(model as Map<String, dynamic>));
+    }
+    //return response.body;
+  }
+  for (final model in restaurantList) {
+    debugPrint(model.toString());
+  }
+  List<int> selectedRestaurantList = [];
+  while (true) {
+    var rnd = Random().nextInt(45) + 1;
```

Comment:
랜덤 max 값에 45를 전달하는데 이것보다는 `foodList.length`를 이용해서 아래와 같이 하는게 좋을듯. 그리고 뒤에 1은 왜 더한거임?
```suggestion
    var rnd = Random().nextInt(foodList.length * 5);
```